flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-starrocks/src/main/java/com/ververica/cdc/connectors/starrocks/sink/StarRocksUtils.java

```diff
@@ -309,7 +315,7 @@ public StarRocksColumn.Builder visit(CharType charType) {
             // varchar type
             int length = charType.getLength();
             long starRocksLength = length * 3L;
-            if (starRocksLength <= MAX_CHAR_SIZE) {
+            if (starRocksLength <= MAX_CHAR_SIZE && !isPrimaryKeys) {
```

Comment:
What's the expected behavior if `starRocksLength` exceeds the `MAX_CHAR_SIZE` for primary keys? throw exception?

Reply:
When the column is a primary key, do not use the char type.

Reply:
In the StarRocks, The primary key columns can be any of the following data types: BOOLEAN, TINYINT, SMALLINT, INT, BIGINT, LARGEINT, STRING, VARCHAR, DATE, and DATETIME. However, the primary key columns cannot be defined as NULL.

Reply:
comments is done